presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayDistinctFromOperator.java

```diff
@@ -47,11 +47,8 @@ public static boolean isDistinctFrom(
             @SqlType("array(E)") Block right,
             @IsNull boolean rightNull)
     {
-        if (leftNull != rightNull) {
-            return true;
-        }
-        if (leftNull) {
-            return false;
+        if (leftNull || rightNull) {
+            return leftNull ^ rightNull;
```

Comment:
You're not right about ^ being bitwise here, it is overloaded for booleans: http://stackoverflow.com/questions/726652/creating-a-logical-exclusive-or-operator-in-java. This is why I preferred it.